docs/src/components/BaseHeader.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+  <vsm-menu ref="header" :menu="menu" v-bind="$attrs">
+    <template #before-nav>
+      <li class="vsm-mob-full">
+        <stripe-logo />
+      </li>
+    </template>
+    <template #default="{ item }">
+      <component :is="item.component" />
+    </template>
```

Comment:
Maybe do something like this:
```vue
<template #default>
    <div>Page 1</div>
    <div>Page 2</div>
    <div>Page 3</div>
</template>
```
or
```vue
<template #default>
    <template>Page 1</template>
    <template>Page 2</template>
    <template>Page 3</template>
</template>
```

```vue
<template #default>
    <template v-for="item in menuItems">{{ item.content }}</template>
</template>
```
This gives the user more control, although I don't know how you would integrate the nav bar and all that custom attribute logic that way.

Reply:
Maybe with a prop or something that holds a menu item object with all the extra info like title, element, etc.